go/applychanges/apply_changes_test.go

```diff
@@ -135,3 +135,43 @@ func TestApplyChangesUUID(t *testing.T) {
 	assert.Equal(t, "02a9920e-b015-4de3-8e32-fb965ac4653c", clay.ID.String())
 	assert.Equal(t, "Clay", clay.Name)
 }
+
+type EmbedStruct struct {
+	ID uuid.UUID `json:"id"`
+}
+type Embedder struct {
+	EmbedStruct
+	Name string `json:"name"`
+}
+type UltimateEmbedder struct {
+	Embedder
+	PowerLevel int `json:"powerLevel"`
+}
+
+func TestApplyChangesOnEmbeddedStructs(t *testing.T) {
+	champion := UltimateEmbedder{
+		PowerLevel: 7900,
+		Embedder: Embedder{
+			Name: "Vegeta",
+			EmbedStruct: EmbedStruct{
+				ID: uuid.Nil,
+			},
+		},
+	}
+
+	happyChanges := map[string]interface{}{
+		"id":         []byte{0x2, 0xa9, 0x92, 0xe, 0xb0, 0x15, 0x4d, 0xe3, 0x8e, 0x32, 0xfb, 0x96, 0x5a, 0xc4, 0x65, 0x3c},
+		"powerLevel": 9001,
+		"name":       "Goku",
+	}
+
+	err := ApplyChanges(happyChanges, &champion)
+	if err != nil {
```

Comment:
Rather than the `if err != nil` pattern, we can use `assert.NoErrorf()`:
```go
err = ApplyChanges(happyChanges, &champion)
assert.NoErrorf(t, err, "ApplyChanges failed")
```

Reply:
Agreed! Thanks Dylan! I corrected this here and anywhere else I found this in the Unit test.

Reply:
Right now it's written as `assert.NoErrorf(t, err, "ApplyChanges failed: %s")` everywhere, which will have an extra `"%s" in the error message since there's no arguments being passed; `Noerrorf` doesn't have any implicit arguments, it logs the error text by default anyways. If I put `return errors.New("Oops")` at the top of `ApplyChanges()`, right now the test output looks like:
```sh
apply_changes_test.go:157: 
                Error Trace:    /home/dylan/source/repos/easi-shared/go/applychanges/apply_changes_test.go:157
                Error:          Received unexpected error:
                                Oops
                Test:           TestApplyChangesOnEmbeddedStructs
                Messages:       ApplyChanges failed: %s
```
That said, this isn't a massive issue or anything, might not be worth the error of removing the `"%s`'s.

Reply:
Thanks @DylanSpOddball, I think that's a good callout, it will be a quick fix,  so I'll implement it.

Reply:
As of https://github.com/CMSgov/easi-shared/pull/13/commits/69cb880fe21db65a750fd18913aae0cb0ce9ddea, looks good.